### PR TITLE
refactor(switch): ts to sfc

### PR DESCRIPTION
**Pull request by Cr0zy07**
### Linked Issues
#492

### CODE REVIEW

File: packages/vue/src/switch/SwitchThumb.vue [hunk 1]
```diff
@@ -29,6 +29,6 @@ const { componentRef } = useComponentRef<HTMLInputElement | null>()
     v-bind="$attrs"
     ref="componentRef"
     :data-state="getState(inject.checked.value)"
-    :data-disabled="inject.disabled ? '' : undefined"
+    :data-disabled="inject.disabled?.value ? '' : undefined"
```
Comment (Cr0zy07) [inline packages/vue/src/switch/SwitchThumb.vue:32]: Could you please validate this, and why it always appears without using `.value` inside the `<template>`

Comment (productdevbook) [reply packages/vue/src/switch/SwitchThumb.vue:32]: yes, I don't know the technical reason behind it but .value sfc adjusts itself. So it doesn't used.

Comment (productdevbook) [reply packages/vue/src/switch/SwitchThumb.vue:32]: inject with .value used. I think the data stays connected up there, otherwise it won't work.

**Review by productdevbook — APPROVED** [review]
LGTM ! ❤️

**Pull request merged**